readthedocs/proxito/middleware.py

```diff
@@ -22,6 +22,59 @@
 log = structlog.get_logger(__name__)  # noqa
 
 
+def _get_domain_from_host(host):
+    """
+    Get the normalized domain from a hostname.
+
+    A hostname can include the port.
+    """
+    return host.lower().split(":")[0]
+
+
+def _unresolve_domain(domain):
+    """
+    Unresolve domain by extracting relevant information from it.
+
+    :param str domain: Domain to extract the information from.
+    :returns: A tuple with: the project slug, domain object, and if the domain
+     is from an external version.
+    """
+    public_domain = _get_domain_from_host(settings.PUBLIC_DOMAIN)
+    external_domain = _get_domain_from_host(settings.RTD_EXTERNAL_VERSION_DOMAIN)
+
+    subdomain, *rest_of_domain = domain.split(".", maxsplit=1)
+    rest_of_domain = rest_of_domain[0] if rest_of_domain else ""
+
+    if public_domain in domain:
+        # Serve from the PUBLIC_DOMAIN, ensuring it looks like `foo.PUBLIC_DOMAIN`.
+        if public_domain == rest_of_domain:
+            project_slug = subdomain
+            return project_slug, None, False
+
+        # TODO: This can catch some possibly valid domains (docs.readthedocs.io.com) for example,
+        # but these might be phishing, so let's ignore them for now.
+        return None, None, False
+
+    if external_domain in domain:
+        # Serve custom versions on external-host-domain.
```

Comment:
Not sure what `custom versions` means here. This is PR builds right? 

```suggestion
        # Serve PR builds on external_domain host.
```